drivers/SmartThings/philips-hue/src/init.lua

```diff
@@ -891,65 +836,35 @@ local stray_bulb_tx, stray_bulb_rx = cosock.channel.new()
+  local thread_local_driver = nil
+
   local process_strays = utils.log_func_wrapper(
-    function(driver, api_instance, strays, bridge_device_uuid)
+    function(driver, strays, bridge_device_uuid)
       local dnis_to_remove = {}
 
-      log.info_with({hub_logs = true}, "Querying bridge for devices from stray light handler")
-      Discovery.search_bridge_for_supported_devices(driver, api_instance, function(hue_driver, svc_info, device_data)
-        if not (svc_info.rid and svc_info.rtype and svc_info.rtype == "light") then return end
-
-        for light_dni, light_device in pairs(strays) do
-          local matching_v1_id = light_device.data and light_device.data.bulbId and
-              light_device.data.bulbId == device_data.id_v1:gsub("/lights/", "")
-          local matching_uuid = light_device.device_network_id == svc_info.rid or
-              light_device.device_network_id == svc_info.rid
-
-          if matching_v1_id or matching_uuid then
-            local api_key_extracted = api_instance.headers["hue-application-key"]
-            log.info_with({ hub_logs = true }, " ", light_device.label, ", re-adding")
-            log.info_with({ hub_logs = true }, string.format(
-              'Found Bridge for stray light %s, retrying onboarding flow.\n' ..
-              '\tMatching v1 id? %s\n' ..
-              '\tMatching uuid? %s\n' ..
-              '\tlight_device DNI: %s\n' ..
-              '\tlight_device Parent Assigned Key: %s\n' ..
-              '\tlight_device parent device id: %s\n' ..
-              '\tProvided bridge_device_id: %s\n' ..
-              '\tCached API key for given bridge_device_id: %s\n' ..
-              '\tCached bridge device for given API key: %s\n'
-              ,
-              light_device.label,
-              matching_v1_id,
-              matching_uuid,
-              light_device.device_network_id,
-              light_device.parent_assigned_child_key,
-              light_device.parent_device_id,
-              bridge_device_uuid,
-              Discovery.api_keys[driver:get_device_info(bridge_device_uuid).device_network_id],
-              driver.api_key_to_bridge_id[api_key_extracted]
-            ))
-            table.insert(dnis_to_remove, light_dni)
-            logged_stray_init(hue_driver, light_device, nil, nil, bridge_device_uuid)
-          end
+      for light_dni, light_device in pairs(strays) do
+        local light_rid = stray_dni_to_rid[light_dni]
+        local cached_light_description = Discovery.light_state_disco_cache[light_rid]
+        if cached_light_description then
+          table.insert(dnis_to_remove, light_dni)
+          logged_migrate_light(driver, light_device, bridge_device_uuid, cached_light_description)
         end
-      end, "[process_strays]")
-      log.info_with({hub_logs = true}, "Finished querying bridge for devices from stray light handler")
+      end
+
       for _, dni in ipairs(dnis_to_remove) do
         strays[dni] = nil
```

Comment:
I'm a little shaky on Lua's pass-by-reference semantics; this *will* mutably update the `stray_lights` map since that's what's beeing passed in, right? Or should this be closing over that stack variable?